openedx/core/djangoapps/programs/utils.py

```diff
@@ -132,28 +133,120 @@ def get_display_category(program):
     return display_candidate
 
 
-def get_engaged_programs(user, enrollments):
-    """Derive a list of programs in which the given user is engaged.
+class ProgramProgressMeter(object):
```

Comment:
I don't like the word "meter". This is a class to calculate the user's program progress. How about "ProgramProgressCalculator"?